web/profiles/custom/os2loop/modules/os2loop_media/modules/os2loop_media_fixtures/src/Fixture/MediaFixture.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Drupal\os2loop_media_fixtures\Fixture;
+
+use Drupal\content_fixtures\Fixture\AbstractFixture;
+use Drupal\content_fixtures\Fixture\DependentFixtureInterface;
+use Drupal\content_fixtures\Fixture\FixtureGroupInterface;
+use Drupal\os2loop_fixtures\Fixture\FileFixture;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\media\Entity\Media;
+
+/**
+ * Media fixture.
+ *
+ * @package Drupal\os2loop_media_fixtures\Fixture
+ */
+class MediaFixture extends AbstractFixture implements DependentFixtureInterface, FixtureGroupInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * MediaFixture constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function load() {
+    $files = $this->entityTypeManager->getStorage('file');
```

Comment:
I think we should get rid of the `FileFixture` class and add a helper function in this class to load fixture files, e.g.

```php
  /**
   * Load all fixture files.
   *
   * @return \Drupal\file\FileInterface[]
   *   The loaded files.
   */
  private function loadFiles(): array {
    $loadedFiles = [];
    $source = __DIR__ . '/../../fixtures/files';
    $files = $this->fileSystem->scanDirectory($source, '/\.(jpg|png|docx|pdf|txt)$/');
    foreach ($files as $file) {
      $name = $file->filename;
      $destination = 'public://fixtures/files/' . $name;
      if (!is_dir(dirname($destination))) {
        $this->fileSystem->mkdir(dirname($destination), 0755, TRUE);
      }
      $loadedFiles[] = file_save_data(
        file_get_contents($file->uri), $destination,
        FileSystemInterface::EXISTS_REPLACE
      );
    }

    return $loadedFiles;
  }
```

and the use it to build `media` fixtures in the `load` method:

```php
  /**
   * {@inheritdoc}
   */
  public function load() {
    $files = $this->loadFiles();
    foreach ($files as $file) {
      $media = Media::create([
        'bundle' => 'os2loop_media_image',
        'field_media_image' => [
          'target_id' => $file->id(),
          'alt' => sprintf('Description of %s.', $file->getFilename()),
        ],
      ]);
      $media->save();
      $this->setReference('media:' . $file->getFilename(), $media);
    }
  }
```